Utils/constants.py

```diff
@@ -0,0 +1,46 @@
+LibDir = "/var/lib/waagent"
+WaagentConf = """\
+#
+# Azure Linux Agent Configuration
+#
+
+Role.StateConsumer=None                 # Specified program is invoked with the argument "Ready" when we report ready status
+                                        # to the endpoint server.
+Role.ConfigurationConsumer=None         # Specified program is invoked with XML file argument specifying role configuration.
+Role.TopologyConsumer=None              # Specified program is invoked with XML file argument specifying role topology.
+
+Provisioning.Enabled=y                  #
+Provisioning.DeleteRootPassword=y       # Password authentication for root account will be unavailable.
+Provisioning.RegenerateSshHostKeyPair=y # Generate fresh host key pair.
+Provisioning.SshHostKeyPairType=rsa     # Supported values are "rsa", "dsa" and "ecdsa".
+Provisioning.MonitorHostName=y          # Monitor host name changes and publish changes via DHCP requests.
+
+ResourceDisk.Format=y                   # Format if unformatted. If 'n', resource disk will not be mounted.
+ResourceDisk.Filesystem=ext4            # Typically ext3 or ext4. FreeBSD images should use 'ufs2' here.
+ResourceDisk.MountPoint=/mnt/resource   #
+ResourceDisk.EnableSwap=n               # Create and use swapfile on resource disk.
+ResourceDisk.SwapSizeMB=0               # Size of the swapfile.
+
+LBProbeResponder=y                      # Respond to load balancer probes if requested by Azure.
+
+Logs.Verbose=n                          # Enable verbose logs
+
+OS.RootDeviceScsiTimeout=300            # Root device timeout in seconds.
+OS.OpensslPath=None                     # If "None", the system default version is used.
+"""
+
+waagent_config_path = "/etc/waagent.conf"
```

Comment:
extensions shouldn't rely on waagent.conf, we should not expose it here so that we do not encourage extensions to take dependencies on it

Reply:
move it inside vmaccess.py instead? We need it to read values like Provisioning.PasswordCryptSaltLength

Reply:
ok, that sounds good (making it local to vmaccess)

it is weird we have this dependency... does vmaccess depend on any other settings? (I'll add comments in the agent's code pointing out this dependency)

Reply:
> move it inside vmaccess.py instead? We need it to read values like Provisioning.PasswordCryptSaltLength

Suggestion, why not only copy the flags that VMAccess needs instead of copying the whole thing. This would remove unnecessary dependencies.

Reply:
@larohra - the problem is that since this was exposed it is now not possible to know if anybody is depending on this. Say (just as an example) somebody created a custom image and set the conf to a specific length, which should be used both by the agent during provisioning and by vmaccess in subsequent password updates.

Reply:
I can only see 2 usages
"Provisioning.PasswordCryptId" and 
"Provisioning.PasswordCryptSaltLength"

Reply:
fixed in [commit](https://github.com/Azure/azure-linux-extensions/pull/1097/commits/89eabc1d5be906fefaa534d1bdcad423297a237e)